accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,87 @@
+  - `TargetFramework` will be included in the folder name if the project is multi-targeted (`TargetFrameworks` is non-empty), or if the `TargetFramework` property was set via the command line (ie is a global property)
+  - `RuntimeIdentifier` will be included in the folder name if it was explicitly set (either in a project file or on the command line).  If it is set automatically by the SDK (for example because `SelfContained` was set), then the `RuntimeIdentifier` will not be included in the folder name
+
+Some examples:
+
+- `artifacts\bin\debug` - The build output path for a simple project when you run `dotnet build`
+- `artifacts\intermediates\debug` - The intermediate output path for a simple project when you run `dotnet build`
+- `artifacts\bin\debug_net8.0` - The build output path for the `net8.0` build of a multi-targeted project
+- `artifacts\publish\release_linux-x64` - The publish path for a simple app when publishing for `linux-x64`
+- `artifacts\package\release` - The folder where the release .nupkg will be created for a project
+
+### Controlling the output path format
+
+We would like the new output format to be used by default.  However, a variety of things may depend on the output path, such as scripts that copy the output of the build, or custom MSBuild logic that hard-codes the output path.  So it would be a breaking change to change the output path format for all projects.
+
+One strategy we often use for situations like this is to make new behavior the default only when targeting the version of .NET that the behavior was introduced in or higher.  This prevents the new .NET SDK from breaking existing, unmodified projects, and ensures that the change in behavior will only be encountered when projects are modified to target a new `TargetFramework`.
+
+The logic for determining which output path format will be used will (mostly) be as follows:
+
+- If a project targets .NET 8 or higher, it will by default use the new output path format.  If it targets .NET 7 or lower, it will by default use the old output path format
+- A project can explicitly choose which output path format to use by setting the `UseArtifactsOutput` property to `true` or `false` in a `Directory.Build.props` file.
+
+### That pesky intermediate output path
+
+Unfortunately, the base intermediate output path can't depend on the project's `TargetFramework`, or anything else in the project file.  The only place to set properties which affect the base intermediate output path is in a `Directory.Build.props` file.  This is because MSBuild imports "project extensions" of the form `<ProjectName>.*.props` and `<ProjectName>.*.targets` from the base intermediate output path, and the project extensions .props files are imported early on in evaluation, before the body of the project file is evaluated.
```

Comment:
What would the value of `MSBuildProjectExtensionsPath` be? 
artifacts?

The assets file does not respect the Configuration pivot.

Reply:
The answer here is that MSBuildProjectExtensionsPath will be artifacts/intermediates